src/main/java/com/es/phoneshop/model/product/cart/DefaultCartService.java

```diff
@@ -48,12 +49,11 @@ public synchronized void addProductToCart(Cart cart, Long productId, int quantit
         }
         CartItem cartItem = new CartItem(product, quantity);
 
-        if (cart.getItems().contains(cartItem)) {
-            CartItem result = cart.getItems().stream()
-                    .filter(cartItem1 -> cartItem1.getProduct().getId().equals(productId))
-                    .findFirst()
-                    .orElseThrow(() -> new ProductNotFoundException(productId));
-            result.setQuantity(result.getQuantity() + quantity);
+        Optional<CartItem> result = cart.getItems().stream()
+                .filter(cartItem1 -> cartItem1.getProduct().getId().equals(productId))
+                .findAny();
+        if (result.isPresent()) {
+            result.get().setQuantity(result.get().getQuantity() + quantity);
             product.setStock(product.getStock() - quantity);
```

Comment:
until order is placed you should not change product stock
just remove this line